server/src/main/java/org/opensearch/plugins/PluginsService.java

```diff
@@ -833,4 +833,8 @@ private String signatureMessage(final Class<? extends Plugin> clazz) {
     public <T> List<T> filterPlugins(Class<T> type) {
         return plugins.stream().filter(x -> type.isAssignableFrom(x.v2().getClass())).map(p -> ((T) p.v2())).collect(Collectors.toList());
     }
+
+    public <T> List<PluginInfo> filterPluginInfos(Class<T> type) {
```

Comment:
qq, why not use the method above and cast the result as list<PluginInfo> ?

Reply:
Good catch. I meant to remove this function.

At one point, I considered switching `IdentityPlugin.getRunAsClient(Plugin plugin)` to `IdentityPlugin.getRunAsClient(PluginInfo pluginInfo)` since the entire instance of a plugin is not necessary here. PluginInfo contains the metadata of a plugin which is sufficient for the IdentityPlugin to use. For the implementation of this interface in the security plugin, the security plugin needs a unique identifier of the plugin. As a convention, the canonical class name will be used as a unique identifier.